### PR TITLE
Add PriorityClass

### PR DESCRIPTION
**Pull request by Vincent056**
Added the ability to set `PriorityClass` through ScanSetting. All the scanner pods will be launched using the `PriorityClass` if set.

### CODE REVIEW

**Comment by xiaojiey** [comment]
@Vincent056 Is it ready for QE test now? Thanks.

**Comment by jhrozek** [comment]
/hold
because this is a feature and there is no clear way to test. I know that PR #50 introduced some docs, but do you think this is enough for QE to verify the feature before merging? 
Same for docs, we should get acks from QE and docs before merging.

**Comment by jhrozek** [comment]
btw I'll review the PR code-wise, but the point still stands

**Comment by Vincent056** [comment]
> 

I will add what's needed for QE and Doc

File: pkg/controller/compliancesuite/suitererunner.go
Comment (Vincent056) [reply pkg/controller/compliancesuite/suitererunner.go:130]: As for right now, we are not validating if there is a priority class exists on the cluster, if the one reference is not there, it will just be ignored, when the pod is launched. I think I can add a warning if the priorityclass referenced was not there.

Comment (Vincent056) [reply pkg/controller/compliancesuite/suitererunner.go:130]: I am going to add validation for this

File: tests/e2e/e2e_test.go
Comment (Vincent056) [reply tests/e2e/e2e_test.go:1851]: that's a good catch, it should be `test-scheduled-suite-priority-class`

File: tests/e2e/e2e_test.go
Comment (Vincent056) [reply tests/e2e/e2e_test.go:1891]: I will add that

**Comment by Vincent056** [comment]
@sheriff-rh  I have added the doc here: https://github.com/ComplianceAsCode/compliance-operator/pull/76,
please let me know if this is good for downstream or if more is needed, thanks!

**Comment by sheriff-rh** [comment]
Hey @Vincent056  thanks for bringing me in. 

I think we have enough information to refer to when adding this to downstream at a later date. Thanks!

File: pkg/controller/compliancesuite/suitererunner.go
Comment (rhmdnd) [inline pkg/controller/compliancesuite/suitererunner.go:29]: We could also say here that the compliance operator is ignoring the priority class and the scan will not be run with a priority class.

File: pkg/controller/compliancesuite/suitererunner.go
Comment (rhmdnd) [inline pkg/controller/compliancesuite/suitererunner.go:95]: Is the `PriorityClass` attribute always present?

File: tests/e2e/helpers.go
Comment (rhmdnd) [inline tests/e2e/helpers.go:2119]: Couldn't we collapse these two assertions into a single function?

Do we need a separate function for checking empty pod priority classes if we have the ability to pass in "" as the priorityClass?

```go
// make sure the priority class matches one of the defaults
expectedPriorityClass := "system-cluster-critical"
waitForPod(checkPodPriorityClass(t, f.KubeClient, pod.Name, namespace, expectedPriorityClass))
// make sure the priority class is empty
expectedPriorityClass := ""
waitForPod(checkPodPriorityClass(t, f.KubeClient, pod.Name, namespace, expectedPriorityClass))
```

Might save us from having a separate, but nearly identical method.

File: CHANGELOG.md
Comment (rhmdnd) [inline CHANGELOG.md:12]: We could make `ScanSetting` a literal.

File: CHANGELOG.md
Comment (rhmdnd) [inline CHANGELOG.md:13]: We could also say this is an optional feature and that scans will continue to work.

File: deploy/compliance-operator-chart/crds/compliance.openshift.io_compliancescans_crd.yaml
Comment (rhmdnd) [inline deploy/compliance-operator-chart/crds/compliance.openshift.io_compliancescans_crd.yaml:80]: After looking through the other description, I think we should avoid the first person point of view (e.g. removing "we").

Perhaps:

```yaml
description: Defines the PriorityClass to use for launching scan-related pods.
  If the PriorityClass is not found, pods won't be scheduled using a PriorityClass.
```

File: deploy/compliance-operator-chart/crds/compliance.openshift.io_compliancesuites_crd.yaml
Comment (rhmdnd) [inline deploy/compliance-operator-chart/crds/compliance.openshift.io_compliancesuites_crd.yaml:102]: Similar comment as above.

File: deploy/compliance-operator-chart/crds/compliance.openshift.io_scansettings_crd.yaml
Comment (rhmdnd) [inline deploy/compliance-operator-chart/crds/compliance.openshift.io_scansettings_crd.yaml:59]: Similar comment as above.

File: deploy/crds/compliance.openshift.io_compliancescans_crd.yaml
Comment (rhmdnd) [inline deploy/crds/compliance.openshift.io_compliancescans_crd.yaml:81]: Similar comment as above.

File: deploy/crds/compliance.openshift.io_compliancesuites_crd.yaml
Comment (rhmdnd) [inline deploy/crds/compliance.openshift.io_compliancesuites_crd.yaml:102]: Similar comment as above.

File: deploy/crds/compliance.openshift.io_scansettings_crd.yaml
Comment (rhmdnd) [inline deploy/crds/compliance.openshift.io_scansettings_crd.yaml:59]: Similar comment as above.

File: doc/crds.md
Comment (rhmdnd) [inline doc/crds.md:609]: Priority class reference, or name? Is there a difference?

**Review by rhmdnd — CHANGES_REQUESTED** [review]
Tested this out locally and it works well! Just some comments on wording and the opportunity to DRY up some of the test code.

Nice work, Vincent.

File: pkg/controller/compliancesuite/suitererunner.go
Comment (Vincent056) [reply pkg/controller/compliancesuite/suitererunner.go:95]: It will be "", if `PriorityClass` is not set

File: doc/crds.md
Comment (Vincent056) [reply doc/crds.md:609]: It should be priorityclass name

File: deploy/compliance-operator-chart/crds/compliance.openshift.io_compliancescans_crd.yaml
Comment (rhmdnd) [inline deploy/compliance-operator-chart/crds/compliance.openshift.io_compliancescans_crd.yaml:81]: I would reword this to remove "we", since we don't use first-person in other parts of the document from what I can tell.

Perhaps:

`this is an optional field, if PriorityClass is invalid or not found, it will be ignored.`

**Review by rhmdnd — COMMENTED** [review]
Looks good, just one minor wording comment.

**Review by rhmdnd — APPROVED** [review]
/lgtm

Thanks, Vincent!

**Comment by rhmdnd** [comment]
Removing the approved the lgtm labels since we still need QE and doc approvals.

**Comment by Vincent056** [comment]
> Removing the approved the lgtm labels since we still need QE and doc approvals.

QE has approved, waiting on the doc approve.

**Comment by rhmdnd** [comment]
Reapplying qe-approved for @xiaojiey 

/label qe-approved

**Review by sheriff-rh — COMMENTED** [review]
Looks good to me! Nice work Vincent.

**Review by rhmdnd — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
